src/NzbDrone.Core/Download/ProcessDownloadDecisions.cs

```diff
@@ -36,52 +38,91 @@ public ProcessedDecisions ProcessDecisions(List<DownloadDecision> decisions)
             var prioritizedDecisions = _prioritizeDownloadDecision.PrioritizeDecisions(qualifiedReports);
             var grabbed = new List<DownloadDecision>();
             var pending = new List<DownloadDecision>();
+            var storedUsenet = new List<DownloadDecision>();
+            var fallbackUsenet = new List<DownloadDecision>();
+            var storedTorrent = new List<DownloadDecision>();
+            var fallbackTorrent = new List<DownloadDecision>();
 
             foreach (var report in prioritizedDecisions)
             {
                 var remoteEpisode = report.RemoteEpisode;
+                var downloadProtocol = report.RemoteEpisode.Release.DownloadProtocol;
 
-                var episodeIds = remoteEpisode.Episodes.Select(e => e.Id).ToList();
-
-                //Skip if already grabbed
-                if (grabbed.SelectMany(r => r.RemoteEpisode.Episodes)
-                                .Select(e => e.Id)
-                                .ToList()
-                                .Intersect(episodeIds)
-                                .Any())
+                // Skip if already grabbed
+                if (IsEpisodeProcessed(grabbed, report))
                 {
                     continue;
                 }
 
                 if (report.TemporarilyRejected)
                 {
-                    _pendingReleaseService.Add(report);
+                    _pendingReleaseService.Add(report, PendingReleaseReason.Delay);
                     pending.Add(report);
                     continue;
                 }
 
-                if (pending.SelectMany(r => r.RemoteEpisode.Episodes)
-                        .Select(e => e.Id)
-                        .ToList()
-                        .Intersect(episodeIds)
-                        .Any())
+                if (IsEpisodeProcessed(pending, report))
                 {
                     continue;
                 }
 
+                if (downloadProtocol == DownloadProtocol.Usenet)
```

Comment:
I'm wondering if we're making this code too complex. It should be possible to simply this.
What if we have the 'grabbed, pending, stored, fallback' lists. And two flags, usenetClientFailed & torrentClientFailed, to substitute the storedUsenet.Any(), then we could write:

```
if (downloadProtocol == DownloadProtocol.Usenet && usenetClientFailed ||
    downloadProtocol == DownloadProtocol.Torrent && torrentClientFailed)
{
    if (IsEpisodeProcessed(stored, report))
    {
        fallback.Add(report);
    }
    else
    {
        stored.Add(report);
    }
}
```

Which is far less verbose, has less lists.

Obviously the usenetClientFailed & torrentClientFailed flags would have to be set appropriately, but that's basically already there.
It still maintains a separate 'path' for usenet vs torrent, as intended originally, and has no behavioral changes afaik.